frontend/packages/kubevirt-plugin/src/components/vms/vm-details.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import { getResource, getServicesForVm } from 'kubevirt-web-ui-components';
+
+import {
+  navFactory,
+  Firehose,
+  StatusBox,
+  ScrollToTopOnMount,
+  SectionHeading,
+} from '@console/internal/components/utils';
+
+import { PodKind, K8sResourceKindReference } from '@console/internal/module/k8s';
+import { PodModel, ServiceModel } from '@console/internal/models';
+import { DetailsPage } from '@console/internal/components/factory';
+
+import { ServicesList } from '@console/internal/components/service';
+import { VMKind, VMIKind } from './types';
+import { VirtualMachineInstanceModel, VirtualMachineInstanceMigrationModel } from '../../models';
+import { VMResourceSummary, VMDetailsList } from './vm-resource';
+
+// import { VmEvents } from './vm-events';
+// import VmConsolesConnected from '../vmconsoles';
+// import { Nic } from '../nic';
+// import { Disk } from '../disk';
+// import { menuActions } from './menu-actions';
+
+export const VMDetailsFirehose = ({ obj: vm }: { obj: VMKind }) => {
+  const { name, namespace } = vm.metadata;
+
+  const vmiRes = getResource(VirtualMachineInstanceModel, {
+    name,
+    namespace,
+    isList: false,
+    prop: 'vmi',
+  });
+  vmiRes.optional = true;
```

Comment:
nit: this could be handled by getResource

Reply:
Right, this is the "enhancement" I wrote about somewhere.
I will put it there but let's not block this PR on that.